test/units/utils/collection_loader/test_collection_loader.py

```diff
@@ -40,6 +48,16 @@ def test_find_module_py3():
     assert path_hook_finder.find_module('missing') is None
 
 
+@pytest.mark.skipif(not PY3 or sys.version_info < (3, 12), reason='Testing Python 2 codepath (find_module) on Python >= 3.12')
```

Comment:
This could be simplified:

```suggestion
@pytest.mark.skipif(sys.version_info < (3, 12), reason='Testing Python 2 codepath (find_module) on Python >= 3.12')
```